# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Для того, чтобы понять, есть ли зависимость возникновения задолженности по кредиту у клиентов от количества детей, мы сгруппируем данные из столбца `debt` по значениям из столбца `children`: посчитаем количество значений - это будет количество заемщиков для каждого количества детей, сумму - т.к.  столбец `debt` у нас имеет тип int64, и состоит только из 0 и 1, то сумма значений покажет нам количество заемщиков с задолженностями, а также посчитаем среднее знаение - это покажет нам долю заемщиков с задолженностью, от общего числа для каждой категории. Для удобства также сделаем дополнительный столбец с процентным количеством заемщиков с задолженностью от общего числа для каждой категории.
Обозначения столбцов:
- `Кол-во детей` - количество детей у клиента;
- `Кол-во заемщиков` - количество клиентов для каждой категории;
- `Кол-во должников` - количество клиентов с задолженностью для каждой категории;
- `Доля с задолженностью` - доля клиентов с задолженностью для каждой категории;
- `% с задолженностью` - процентное соотношение людей с задолженностью от всех клиентов для каждой категории.

In [22]:
#группируем данные из столбца 'debt' по 'children', считаем их количество, сумму, среднее значение
data_groupe_children = data.groupby(['children'], as_index=False).agg({'debt': ['count', 'sum', 'mean']})
#создаем новый столбец mean_percent, в котором, вместо долей будут проценты
data_groupe_children['mean_percent']  = data_groupe_children['debt']['mean'] * 100
#переименовываем столбцы
data_groupe_children = data_groupe_children.rename(columns={'children':'Кол-во детей', 
                                                            'debt':' ', 
                                                            'count':'Кол-во заемщиков', 'sum':'Кол-во должников',
                                                            'mean':'Доля с задолженностью', 
                                                            'mean_percent':'% с задолженностью'})
#выводим таблицу
data_groupe_children


Кол-во детей                                                          \
               Кол-во заемщиков Кол-во должников Доля с задолженностью   
0            0            14091             1063              0.075438   
1            1             4808              444              0.092346   
2            2             2052              194              0.094542   
3            3              330               27              0.081818   
4            4               41                4              0.097561   
5            5                9                0              0.000000   

  % с задолженностью  
                      
0           7.543822  
1           9.234609  
2           9.454191  
3           8.181818  
4           9.756098  
5           0.000000

Мы видим, что в наших данных всего 41 человек, у которых 4 детей, и 9 человек, с 5 детьми. Поэтому выводы об этих катеориях по этим данным, возможно, будут ошибочны. В итоговом выводе мы их не будем учитывать.

**Вывод:** мы видим, что по сравнению с людьми без детей:
люди с 1 ребенком на 1,69% чаще не платят по кредитам в срок;
люди с 2 детьми на 1,91% чаще не платят по кредитам в срок;
люди с 3 детьми на 0,63% чаще не платят по кредитам в срок;
на основании этого, мы можем предположить, что наличие ребенка увеличивает риск того, что человек заплатит по кредиту в срок.

Если мы посчитаем, что данных для клиентов с 3 детьми тоже недостаточно, для вывода, то мы можем предположить, что с увеличеснием количества детей у клиента он с большей вероятностью будет иметь задолженность по кредиту.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Посмотрим, какие уникальные значения у нас указаны в столбце `family_status`:

In [23]:
data['family_status'].unique()

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'Не женат / не замужем'], dtype=object)

Есть 5 категорий.

Как и в предыдущем случаем, мы сгруппируем данные из столбца `debt`, но уже по столбцу `family_status`. Посчитаем их количество, количество заемщиков с задолженностью, их долю, а также % соотношение.
Обозначения столбцов:
- `Семейное положение` - Семейное положение у клиента;
- `Кол-во заемщиков` - количество клиентов для каждой категории;
- `Кол-во должников` - количество клиентов с задолженностью для каждой категории;
- `Доля с задолженностью` - доля клиентов с задолженностью для каждой категории;
- `% с задолженностью` - процентное соотношение людей с задолженностью от всех клиентов для каждой категории.

In [24]:
#группируем данные из столбца 'debt' по 'children', считаем их количество, сумму, среднее значение
data_groupe_family_status = data.groupby(['family_status'], as_index=False).agg({'debt': ['count', 'sum', 'mean']})
#создаем новый столбец 'mean_percent', в котором, вместо долей, будут проценты
data_groupe_family_status['mean_percent']  = data_groupe_family_status['debt']['mean'] * 100
#переименовываем столбцы
data_groupe_family_status = data_groupe_family_status.rename(columns={'family_status':'Семейное положение', 
                                                            'debt':' ', 
                                                            'count':'Кол-во заемщиков', 'sum':'Кол-во должников',
                                                            'mean':'Доля с задолженностью', 
                                                            'mean_percent':'% с задолженностью'})
#выводим таблицу
data_groupe_family_status


Семейное положение                                    \
                         Кол-во заемщиков Кол-во должников   
0  Не женат / не замужем             2796              273   
1              в разводе             1189               84   
2         вдовец / вдова              951               63   
3       гражданский брак             4134              385   
4        женат / замужем            12261              927   

                        % с задолженностью  
  Доля с задолженностью                     
0              0.097639           9.763948  
1              0.070648           7.064760  
2              0.066246           6.624606  
3              0.093130           9.313014  
4              0.075606           7.560558

Мы видим, что в каждой катеогрии достаточно данных, чтобы сделать выводы.

**Вывод:** Мы можем предположить, что вдовы и вдовцы своевременнее всех платят по кредитам, в то время, как люди, никогда не состоящие в браке чаще имеют проблемы со своевременностью оплаты по кредиту.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Для более удобного сравнения клиентов по уровню дохода разделим их на 5 равных групп. Для определения группы поделим наши данные на равные части методом qcut():

In [25]:
pd.qcut(data['total_income'], q=5)

0        (214604.0, 2265604.0]
1          (98514.0, 132113.0]
2         (132113.0, 161380.0]
3        (214604.0, 2265604.0]
4         (132113.0, 161380.0]
                 ...          
21520    (214604.0, 2265604.0]
21521     (132113.0, 161380.0]
21522     (20666.999, 98514.0]
21523    (214604.0, 2265604.0]
21524     (20666.999, 98514.0]
Name: total_income, Length: 21331, dtype: category
Categories (5, interval[float64]): [(20666.999, 98514.0] < (98514.0, 132113.0] < (132113.0, 161380.0] < (161380.0, 214604.0] < (214604.0, 2265604.0]]

Данные поделлились на 4 категории, назовем их для удобства:
- 0–98514 — `'E'`;
- 98515–132113 — `'D'`;
- 132114–161380 — `'C'`;
- 161381–214604 — `'B'`;
- 214605 и выше — `'A'`.

и добавим название для каждой категории в столбцец `income_type`, а значения в столбец `income_range`:

In [26]:
#создаем список названий для категорий
income_lebel = ['E', 'D', 'C', 'B', 'A']
#добавляем столбец 'income_range' со значениями категорий дохода
data['income_range'] = pd.qcut(data['total_income'], q=5)
#добавляем столбец 'income_type' со названиями категорий дохода
data['income_type'] = pd.qcut(data['total_income'], q=5, labels=income_lebel)
#выводим первые 5 строк
data.head()


children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F           A     0        253875   
1   женат / замужем                 0      F           D     0        112080   
2   женат / замужем                 0      M           C     0        145885   
3   женат / замужем                 0      M           A     0        267628   
4  гражданский брак                 1      F           C     0        158616   

                      purpose total_income_category          purpose_category  \
0               покупка жилья                     B  операции с недвижимостью   
1     приобретение автомобиля                     C    операции с автомобилем   
2               покупка жилья                     C  операции с недвижимостью   
3  дополнительное образование                     B     получение образования   
4             сыграть свадьбу                     C        проведение свадьбы   

            income_range  
0  (214604.0, 2265604.0]  
1    (98514.0, 132113.0]  
2   (132113.0, 161380.0]  
3  (214604.0, 2265604.0]  
4   (132113.0, 161380.0]

Проверим, ровно ли поделились значения:

In [27]:
data['income_type'].value_counts()

E    4267
D    4266
C    4266
B    4266
A    4266
Name: income_type, dtype: int64

Данные поделились на раыне категории.

Как и в предыдущих случаях, мы сгруппируем данные из столбца `debt`, но уже по столбцам `income_range`  и `income_type`. Посчитаем их количество, количество заемщиков с задолженностью, их долю, а также % соотношение.
Обозначения столбцов:
- `Уровень дохода` - Диапазон уровня дохода;
- `Категория` - Категория уровня дохода;
- `Кол-во заемщиков` - количество клиентов для каждой категории;
- `Кол-во должников` - количество клиентов с задолженностью для каждой категории;
- `Доля с задолженностью` - доля клиентов с задолженностью для каждой категории;
- `% с задолженностью` - процентное соотношение людей с задолженностью от всех клиентов для каждой категории.


In [28]:
#группируем данные из столбца 'debt' по 'children', считаем их количество, сумму, среднее значение, убираем пустые значения и сбрасываем индексы
data_group_income = data.groupby(['income_range','income_type']).agg({'debt': ['count', 'sum', 'mean']}).dropna().reset_index()
#создаем новый столбец 'mean_percent', в котором, вместо долей, будут проценты
data_group_income['mean_percent']  = data_group_income['debt']['mean'] * 100
#переименовываем столбцы
data_group_income= data_group_income.rename(columns = {'income_range':'Уровень дохода', 
                                                       'income_type':'Категория',
                                                      'debt':' ',
                                                      'count':'Кол-во заемщиков',
                                                      'sum':'Кол-во должников',
                                                      'mean':'Доля с задолженностью',
                                                      'mean_percent':'% с задолженностью'})
#выводим таблицу
data_group_income

Уровень дохода Категория                                    \
                                   Кол-во заемщиков Кол-во должников   
0   (20666.999, 98514.0]         E             4267              344   
1    (98514.0, 132113.0]         D             4266              358   
2   (132113.0, 161380.0]         C             4266              373   
3   (161380.0, 214604.0]         B             4266              358   
4  (214604.0, 2265604.0]         A             4266              299   

                        % с задолженностью  
  Доля с задолженностью                     
0              0.080619           8.061870  
1              0.083919           8.391936  
2              0.087436           8.743554  
3              0.083919           8.391936  
4              0.070089           7.008908

**Вывод:** На основании данных мы можем предположить, что с ростом дохода количество должников растет до определенного уровня (до дохода в диапазоне 132113 - 161381), а потом начинает снижаться. Меньше всего должников с доходом от 214605. 

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Как и в предыдущем случаем, мы сгруппируем данные из столбца `debt`, но уже по столбцу `family_status`. Посчитаем их количество, количество заемщиков с задолженностью, их долю, а также % соотношение.
Обозначения столбцов:
- `Цель кредита` - Цель, на которую брался кредит;
- `Кол-во заемщиков` - количество клиентов для каждой категории;
- `Кол-во должников` - количество клиентов с задолженностью для каждой категории;
- `Доля с задолженностью` - доля клиентов с задолженностью для каждой категории;
- `% с задолженностью` - процентное соотношение людей с задолженностью от всех клиентов для каждой категории.

In [29]:
#группируем данные из столбца 'debt' по 'children', считаем их количество, сумму, среднее значение
data_groupe_purpose = data.groupby(['purpose_category'], as_index=False).agg({'debt': ['count', 'sum', 'mean']})
#создаем новый столбец 'mean_percent', в котором, вместо долей, будут проценты
data_groupe_purpose['mean_percent']  = data_groupe_purpose['debt']['mean'] * 100
#переименовываем столбцы
data_groupe_purpose = data_groupe_purpose.rename(columns={'purpose_category':'Цель кредита', 
                                                            'debt':' ', 
                                                            'count':'Кол-во заемщиков', 'sum':'Кол-во должников',
                                                            'mean':'Доля с задолженностью', 
                                                            'mean_percent':'% с задолженностью'})
#выводим таблицу
data_groupe_purpose


Цель кредита                                    \
                            Кол-во заемщиков Кол-во должников   
0    операции с автомобилем             4279              400   
1  операции с недвижимостью            10751              780   
2     получение образования             3988              369   
3        проведение свадьбы             2313              183   

                        % с задолженностью  
  Доля с задолженностью                     
0              0.093480           9.347978  
1              0.072551           7.255139  
2              0.092528           9.252758  
3              0.079118           7.911803

Мы видим, что в каждой катеогрии достаточно данных, чтобы сделать выводы.

**Вывод:** Люди, котрые берут кредиты на операции с недвижимостью, реже всех имеют задолженности по кредиту, в то время, как люди, берущие кредит на операции с автомобилем, чаще других имеют задолженности по кредиту. 

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Пропуски в данных у нас были в столбцах total_income и days_employed. И количество пропусков было одинаковым - по 2174. И либо одинаковое количество людей не захотели считать или расскрывать свой трудовой стаж и доход, что маловероятно для закредитованных клиентов банка, либо была какая-то техническая ошибка при сборе или выгрузки данных.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* т.к. на медианные значения слабее влияют выбросы, как, например, на среднее арифметические. При нормальном распределении, можно было бы использовать средне арифметическое значение, но т.к. у нас были данные с явными выбросами (например 340266.072047 дней трудового стажа в 4 строке, что равно 932 годам), оно бы показывало искаженные значения.

### Шаг 4: общий вывод.

В данной работе мы работали со статистикой о платёжеспособности 21525 клиентов, в которой были указаны:
-  количество детей в семье
-  общий трудовой стаж в днях
-  возраст клиента в годах
-  уровень образования клиента
-  семейное положение
-  пол клиента
-  тип занятости
-  имел ли задолженность по возврату кредитов
-  ежемесячный доход
-  цель получения кредита.

У 2174 клиентов в `общий трудовой стаж в днях` и `ежемесячный доход` были пропуски в данных. Это значит, что либо одинаковое количество людей не захотели считать или расскрывать свой трудовой стаж и доход, что маловероятно для закредитованных клиентов банка, либо была какая-то техническая ошибка при сборе или выгрузки данных, которую рекомендуем в дальнейшем исправить. Но в данном исследовании мы заполнили пропуски в ужемесечном доходе и общем трудовом стаже по медианным значениям для каждого типа занятости.

Также при анализе обнаружились аномальные данные общего трудового стажа у некоторых клиентов по типу занятости, а именно у безработных и пенсионеров. В данном исследовании эти данные не учитывались.

И в данных были обнаружены 71 дубликат, которые мы не учитывали в исследовании. 

По итогам исследования мы можем предположить:
- есть зависимость между количеством детей и возвратом кредита в срок:  с увеличеснием количества детей у клиента он с большей вероятностью будет иметь задолженность по кредиту.
- есть зависимость, между семейным положением и возвратом редита в срок: вдовцы своевременнее всех платят по кредитам, в то время, как люди, никогда не состоящие в браке чаще имеют проблемы со своевременностью оплаты по кредиту.
- есть зависимость между уровнем дохода и возвратом кредита в срок: с ростом дохода количество должников растет до определенного уровня (до дохода в диапазоне 132113 - 161381), а потом начинает снижаться. Меньше всего должников с доходом от 214605.
- есть зависимость между уровнем дохода и возвратом кредита в срок: клиенты, котрые берут кредиты на операции с недвижимостью, реже всех имеют задолженности по кредиту, в то время, как люди, берущие кредит на операции с автомобилем, чаще других имеют задолженности по кредиту.